code/headers/coordinate_3D_c.hpp

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+/**
+ * Class point_3d is used for xyz 3D coordinates
+ */
+
+namespace r2d2::robot_arm {
+    class coordinate_3D_c {
+    public:
+        /**
+         * Operator == for equality checking.
+         * Required for catch2.
+         *
+         * @param coordinate_3D_c rhs
+         * @return bool
+         **/
+        bool operator==(const coordinate_3D_c &rhs) const;
```

Comment:
Why is this operator not with the rest of the operators?

Reply:
So that it is easy to find for potential removal later. The reason this operator exists is because it simplified the process of error-checking in the catch2 library.